arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/floorfilter/SiteFacilityView.kt

```diff
@@ -0,0 +1,484 @@
+            updateSiteFacilityTitle()
+
+            dialog.setOnDismissListener {
+                closeSiteFacilityView()
+            }
+            this.dialog = dialog
+        }
+    }
+
+    internal fun close() {
+        val dialog = this.dialog ?: return
+        dialog.dismiss()
+        (parent as? ViewGroup)?.removeView(this)
+        this.dialog = null
+    }
+
+    /**
+     * Shows or hides the search bar based on [UiParameters.hideFacilitySearch]
+     * and [UiParameters.hideSiteSearch].
+     *
+     * @since 100.13.0
+     */
+    internal fun showHideSearch() {
+        val hideSearch = if (isShowingFacilities) {
+            uiParameters.hideFacilitySearch
+        } else {
+            uiParameters.hideSiteSearch
+        }
+        siteFacilitySearchLayout?.visibility = if (hideSearch) View.GONE else View.VISIBLE
+        siteFacilityTitleSeparator?.visibility = if (hideSearch) View.VISIBLE else View.GONE
+    }
+
+    /**
+     * Handles changes to the configurable UI parameters.
+     *
+     * @since 100.13.0
+     */
+    internal fun processUiParamUpdate() {
+        setBackgroundColor(uiParameters.buttonBackgroundColor)
+        siteFacilitySearchLayout?.setBackgroundColor(uiParameters.searchBackgroundColor)
+        siteFacilitySearchEditText?.setBackgroundColor(uiParameters.searchBackgroundColor)
+
+        uiParameters.setButtonTintColors(siteFacilityBackButton)
+        backToSitesButtonSeparator?.setBackgroundColor(uiParameters.closeButtonBackgroundColor)
+        siteFacilityTitleSeparator?.setBackgroundColor(uiParameters.closeButtonBackgroundColor)
+        uiParameters.setButtonTintColors(siteFacilityCloseButton)
+        uiParameters.setButtonTintColors(siteFacilitySearchClearButton)
+        uiParameters.setButtonTintColors(siteFacilitySearchButton)
+
+        siteFacilityTitle?.setTextColor(uiParameters.textColor)
+        siteFacilitySubtitle?.setTextColor(uiParameters.textColor)
+        siteFacilitySearchEditText?.setHintTextColor(uiParameters.textColor)
+        siteFacilitySearchEditText?.setTextColor(uiParameters.textColor)
+        siteFacilityEmptyView?.setTextColor(uiParameters.textColor)
+
+        uiParameters.setScrollbarColor(siteFacilityRecyclerView)
+
+        siteFacilityAdapter.notifyDataSetChanged()
+    }
+
+    private fun clearSearch() {
+        siteFacilitySearchEditText?.setText("")
+    }
+
+    private fun getSelectedSite(): FloorSite? {
+        return floorFilterManager?.getSelectedSite()
+    }
+
+    private fun isSelectedSite(site: FloorSite): Boolean {
+        return floorFilterManager?.isSelectedSite(site) ?: false
+    }
+
+    private fun getSelectedFacility(): FloorFacility? {
+        return floorFilterManager?.getSelectedFacility()
+    }
+
+    private fun isSelectedFacility(facility: FloorFacility): Boolean {
+        return floorFilterManager?.isSelectedFacility(facility) ?: false
+    }
+
+
+    private var siteFacilityBackButton: ImageView? = null
```

Comment:
please put the properties before the constructors and init block.

_The contents of a class should go in the following order:
Property declarations and initializer blocks
Secondary constructors
Method declarations
Companion object_

https://kotlinlang.org/docs/coding-conventions.html#class-layout